src/middlewares/essence/CollectSealedAuctionMw.sol

```diff
@@ -0,0 +1,392 @@
+        _allBids[bidCounter] = newBid;
+        _bidders[namespace][profileId][essenceId].push(newBid);
+
+        emit BidPlaced(
+            bidCounter,
+            collector,
+            amount,
+            profileId,
+            essenceId,
+            namespace
+        );
+    }
+
+    function withdraw(uint256 profileId, uint256 essenceId) public {
+        require(
+            0 < _data[namespace][profileId][essenceId].totalSupply,
+            "INVALID_PROFILE_ESSENCE"
+        );
+        require(
+            block.timestamp >
+                _data[namespace][profileId][essenceId].endTimestamp,
+            "NOT_ENDED"
+        );
+        uint256[] memory topBids = findTopXBidders(
+            _data[namespace][profileId][essenceId].totalSupply,
+            profileId,
+            essenceId
+        );
+
+        bool canWithdraw;
+        for (
+            uint256 i = 0;
+            i < _bidders[namespace][profileId][essenceId].length;
+            i++
+        ) {
+            if (
+                _bidders[namespace][profileId][essenceId][i].bidder ==
+                msg.sender &&
+                _bidders[namespace][profileId][essenceId][i].collected == false
+            ) {
+                bool flag;
+                for (uint256 j = 0; j < topBids.length; j++) {
+                    if (
+                        _bidders[namespace][profileId][essenceId][i].id ==
+                        topBids[j]
+                    ) {
+                        flag = true;
+                    }
+                }
+                if (!flag) {
+                    canWithdraw = true;
+                    IERC20(_data[namespace][profileId][essenceId].currency)
+                        .safeTransfer(
+                            msg.sender,
+                            _bidders[namespace][profileId][essenceId][i].amount
+                        );
+                    _bidders[namespace][profileId][essenceId][i]
+                        .collected = true;
+                }
+            }
+        }
+        require(canWithdraw, "CANNOT_WITHDRAW");
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              INTERNAL
+    //////////////////////////////////////////////////////////////*/
+
+    function _checkSubscribe(
+        address _namespace,
+        uint256 profileId,
+        address collector
+    ) internal view returns (bool) {
+        address essenceOwnerSubscribeNFT = IProfileNFT(_namespace)
+            .getSubscribeNFT(profileId);
+
+        return (essenceOwnerSubscribeNFT != address(0) &&
+            IERC721(essenceOwnerSubscribeNFT).balanceOf(collector) > 0);
+    }
+
+    function _checkProfile(address _namespace, address collector)
+        internal
+        view
+        returns (bool)
+    {
+        return (IERC721(_namespace).balanceOf(collector) > 0);
+    }
+
+    // function to find top bidders
+    function findTopXBidders(
```

Comment:
This method looks needs very high gas. Could you please write a test that if there are 3000 bidders, how much gas it would cost?